api/v1/types/util.go

```diff
@@ -63,3 +69,23 @@ func (p *GetLayerStatsActiveAccountsParams) Validate() error {
 	}
 	return &InvalidParamFormatError{ParamName: "window_step_seconds", Err: fmt.Errorf("invalid value: %d", *p.WindowStepSeconds)}
 }
+
+// The autogenerated server performs type validation that the input string
+// conforms to either an oasis bech32-encoded address or an ethereum
+// address, but we double check just to be sure.
+func UnmarshalToOcAddress(raw *string) (*StakingAddress, error) {
+	if raw == nil {
+		return nil, nil
+	}
+	if ethCommon.IsHexAddress(*raw) {
+		ea := ethCommon.HexToAddress(*raw)
+		ctx := sdkTypes.AddressV0Secp256k1EthContext
+		oasisAddr := StakingAddress(address.NewAddress(ctx, ea[:]))
```

Comment:
is this cast right?

Reply:
Think so; they're the same underlying type but `StakingAddress` is the type used in this `api` package. Unless I'm missing something?

Reply:
leme just try this, I thought maybe this would cast the bytes-like address into the string-like address via treating it like UTF-8, without going through the bech32 stuff

Reply:
ok you're right, StakingAddress is not the string-like thing

Reply:
for my information:

- address.NewAddress here is from the oasis-core/go/common/crypto/address package where `type Address [Size]byte`
- StakingAddress is from the generated API types where `type StakingAddress = staking.Address` (alias declaration)
  - `staking` here is our vendored Damask staking/api package where `type Address = original.Address` (alias declaration)
    - `original` here is oasis-core/go/staking/api where `type Address address.Address` (new type definition)
      - `address` here is oasis-core/go/common/crypto/address